### ISSUE TITLE
`poetry show --tree` is very slow when the output is large

### ISSUE THREAD

**Issue by dimbleby** [sdispater/poetry#1298]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **Centos 7**: 
- **0.12.17**: 
- **Link of a [Gist](https://gist.github.com/dimbleby/8353512c39a8222f49a8a3c1a1c84a3e) with the contents of your pyproject.toml file**: 

## Issue

Running `poetry show --tree` for this `pyproject.toml` starts off well, but then grinds gradually to a near halt.  

See this demonstration: https://asciinema.org/a/H1Smy0VKzs7TKIFAYy9I1aO7t.  Bigger projects go even slower.

Profiling, as `python -m cProfile -s time $(which poetry) show --tree`, suggests that it's to do with the coloring:

```
         12370463 function calls (12344456 primitive calls) in 71.880 seconds

   Ordered by: internal time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1412   56.347    0.040   66.413    0.047 pastel.py:67(colorize)
     1128    2.398    0.002    2.398    0.002 buffered_output.py:21(do_write)
   610230    2.366    0.000    3.541    0.000 style.py:123(apply)
   610238    1.505    0.000    5.989    0.000 pastel.py:145(_apply_current_style)
```

Perhaps this bug report should go in https://github.com/sdispater/pastel?  But from a user point of view, it's certainly `poetry` that's slow.

**Comment by dimbleby** [comment]
I've done some digging, and I'm suspicious that the issue is in [cleo](https://github.com/sdispater/cleo/), specifically [here](https://github.com/sdispater/cleo/blob/98dd69fb7ddc1f562a69bbbf6f03fc02d9fbc436/cleo/styles/cleo_style.py#L246-L250).

This looks to me as though it doesn't "preserve the last four characters inserted" but rather preserves everything _except_ the last four characters - though I confess that I haven't really understood what this code is doing.

Nonetheless, what I see is that the `_buffered_output` here grows large; and that if I change the code so that it goes `m[-4:]` instead of `m[:-4]` - then it's a *lot* faster.

**Comment by sdispater** [comment]
Thanks for taking the time to investigate this issue.

Cleo has been modified heavily for the 0.7.0 release (and now depends mostly on [clikit](https://github.com/sdispater/clikit)) and the issue is fixed in the 1.0.0b1 release of Poetry
